src/models/PlantModel.ts

```diff
@@ -1,74 +1,83 @@
-import { HandleFile, FileType } from './HandleFile';
-import { IPlant, IOpsInfo } from '../interfaces';
+import { ResultSetHeader } from 'mysql2';
+import connection from './connection';
+import { IPlant } from '../interfaces';
 
 class PlantModel {
-  private handleFile = new HandleFile();
-
-  private async updateOpsInfo(incrementAmount = 1): Promise<number> {
-    const opsInfo = await this.handleFile.readFile<IOpsInfo>(FileType.OpsInfo);
-    opsInfo.createdPlants += incrementAmount;
-
-    await this.handleFile.saveFile(FileType.OpsInfo, opsInfo);
-
-    return opsInfo.createdPlants;
-  }
+  private conn = connection;
 
   public async getAll(): Promise<IPlant[]> {
-    const plants = await this.handleFile.readFile<IPlant[]>(FileType.Plants);
+    const query = `SELECT
+      id, breed, size, needs_sun as needsSun, origin, water_Frequency as waterFrequency
+      FROM plants`;
+    const [rows] = await this.conn.execute(query);
```

Comment:
O que acha de passar o generic na hora de chamar a função? Como se usa o `as` depois, não tem diferença inicialmente, mas assim a gente reforça a utilização do generic (que não está sendo inferido)

```suggestion
    const [rows] = await this.conn.execute<RowDataPacket[]>(query);
```

Reply:
Alteração implementada